compiler/hie-core/src/Development/IDE/Core/Shake.hs

```diff
@@ -154,6 +151,25 @@ instance Hashable Key where
 --   not propagate diagnostic errors through multiple phases.
 type IdeResult v = ([FileDiagnostic], Maybe v)
 
+data Value v
+    = Succeeded v
+    | Failed
```

Comment:
I suppose the plan is to add a third alternative here for stale values?

Reply:
Exactly.